src/features/EditMoviePage/EditMoviePage.jsx

```diff
@@ -0,0 +1,161 @@
+import React, { useState } from 'react';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+
+import { propTypesShapes } from '../../constants';
+import CustomInput from '../../components/CustomInput/CustomInput';
+import CustomTextarea from '../../components/CustomTextarea/CustomTextarea';
+import CustomButton from '../../components/CustomButton/CustomButton';
+import { editMovie } from '../../redux/movies/movies.actions';
+import { moviesDataSelector } from '../../redux/movies/movies.selectors';
+
+import styles from './EditMoviePage.module.scss';
+
+
+const EditMoviePage = ({ match, movies, editMovie, history }) => {
+  const editableMovieId = Number(match.params.id);
+
+  const { id, title, posterUrl, director, genres, description } = movies.find((movie) => movie.id === editableMovieId);
+
+  const INITIAL_DATA = {
+    title,
+    posterUrl,
+    director,
+    genres,
+    description,
+  };
+
+  const [formData, changeFormData] = useState(INITIAL_DATA);
+
+  const [errors, setErrors] = useState([]);
+
+  const inputChangeHandler = (event, fieldName) => {
+    const value = event.target.value;
+    changeFormData((prevState) => ({
+      ...prevState,
+      [fieldName]: value.trim()
+    }))
+  };
+
+  const genresChangeHandler = (event) => {
+    const inputValue = event.target.value;
+    const genreRegExp = /[A-Za-z\s]{3,}/gm;
+
+    changeFormData((prevState) => ({
+      ...prevState,
+      genres: inputValue.match(genreRegExp)
+    }));
+  };
+
+  const validateForm = (formData) => {
+    const validationResult = {
+      isValid: true,
+      errors: []
+    };
+
+    if (formData.title.length < 3) {
+      validationResult.errors.push('Invalid title.')
+    }
+
+    const urlRegExp = /(http(s)?:\/\/.)[-a-zA-Z0-9%:._]{2,256}\.[a-z]{2,6}\b([-a-zA-Z0-9_.?&/]*)/gm;
+    if (!urlRegExp.test(formData.posterUrl)) {
+      validationResult.errors.push('Invalid poster url.');
+    }
+
+    if (validationResult.errors.length) {
+      validationResult.isValid = false;
+    }
+
+    return validationResult;
```

Comment:
1. better to use if .. else if ... else ... here
2. You don't need isValid flag here, you can simplify and rely on validationResult.error.length